smurfs/src/reducers/index.js

```diff
@@ -21,3 +21,47 @@
   There is no need for 'combineReducers' in this project.
   Components can then read your store as, `state` and not `state.fooReducer`.
 */
+import {SMURF_START,SMURF_SUCCESS,SMURF_FAIL,ADD_SMURF,ADD_SMURF_FAIL} from '../actions/index'
+
+const initialState={
+  fetching:false,
+  smurfs:[],
+  error:null
+  
+}
+
+export const rootReducer = (state = initialState, action) => {
+  switch (action.type) {
+
+    case SMURF_START:
+    return{
+      ...state,
+      fetching:true,
+     error:null
+      }
+    case SMURF_SUCCESS:
+    return{
+      ...state,
+      
+      
+      smurfs:[...state.smurfs,...action.payload],
```

Comment:
This works just fine in your app, but if for some reason you fetched smurf data again it would duplicate your array, it's probably better to just use
```
smurf: action.payload
```